src/Telegram.Bot/ITelegramBotClient.cs

```diff
@@ -875,15 +875,47 @@ public interface ITelegramBotClient
         /// <param name="parseMode">Send Markdown or HTML, if you want Telegram apps to show bold, italic, fixed-width text or inline URLs in the media caption.</param>
         /// <param name="replyMarkup">A JSON-serialized object for an inline keyboard.</param>
         /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
-        /// <returns><c>true</c> on success.</returns>
-        /// <see href="https://core.telegram.org/bots/api#editmessagecaption"/>
+        /// <returns>On success, the edited Description is returned.</returns>
+        /// <see href="https://core.telegram.org/bots/api#editmessagetext"/>
         Task EditMessageCaptionAsync(
             string inlineMessageId,
             string caption,
             InlineKeyboardMarkup replyMarkup = default,
             CancellationToken cancellationToken = default,
             ParseMode parseMode = default);
 
+        /// <summary>
+        /// Use this method to edit audio, document, photo, or video messages.
+        /// </summary>
+        /// <param name="chatId"><see cref="ChatId"/> for the target chat</param>
+        /// <param name="messageId">Unique identifier of the sent message</param>
+        /// <param name="media">A JSON-serialized object for a new media content of the message</param>
+        /// <param name="replyMarkup">A JSON-serialized object for an inline keyboard.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <returns>On success the edited <see cref="Message"/> is returned.</returns>
+        /// <see href="https://core.telegram.org/bots/api#editmessagemedia"/>
+        Task<Message> EditMessageMediaAsync(
+            ChatId chatId,
+            int messageId,
+            InputMedia media,
+            InlineKeyboardMarkup replyMarkup = default,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Use this method to edit audio, document, photo, or video messages.
+        /// </summary>
+        /// <param name="inlineMessageId">Unique identifier of the sent message</param>
+        /// <param name="media">A JSON-serialized object for a new media content of the message</param>
+        /// <param name="replyMarkup">A JSON-serialized object for an inline keyboard.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <returns>On success the edited <see cref="Message"/> is returned.</returns>
+        /// <see href="https://core.telegram.org/bots/api#editmessagemedia"/>
+        Task<Message> EditMessageMediaAsync(
```

Comment:
Same, it should return `Task` without `Message` since it's inline version.

Reply:
Can we also add overload method with correct argument order - `CancellationToken cancellationToken` as  the last parameter?

Reply:
Argument order was changed by @pouladpld in [this commit](https://github.com/TelegramBots/Telegram.Bot/commit/f58b5d5893200228ad65eb70b58cc6e14ee94959) to avoid a breaking change

Reply:
Speaking of which, is there any way to avoid doing the same with all the other newly added parameters?

Reply:
we can either introduce new update with breaking changes or add two implementations for each request - one with recommended parameter order and the other as it is now

Reply:
@karb0f0s sounds good. we can use `[Obsolete]` to deprecate the overloaded methods with non-consistent param list.